mula/scheduler/schedulers/schedulers/report.py

```diff
@@ -87,6 +87,30 @@ def push_tasks_for_rescheduling(self):
         ) as executor:
             for schedule in schedules:
                 report_task = ReportTask.model_validate(schedule.data)
+
+                # When the schedule has no schedule, but a task is
+                # already executed we should not push the task again
+                task_db = None
+                try:
+                    task_db = self.ctx.datastores.task_store.get_latest_task_by_hash(report_task.hash)
+                    if task_db and schedule.schedule is None:
+                        self.logger.debug(
+                            "Schedule has no schedule, but task already executed",
+                            schedule_id=schedule.id,
+                            scheduler_id=self.scheduler_id,
+                            organisation_id=self.organisation.id,
+                        )
+                        continue
+                except storage.errors.StorageError as exc_db:
+                    self.logger.error(
+                        "Could not get latest task by hash %s",
+                        report_task.hash,
+                        scheduler_id=self.scheduler_id,
+                        organisation_id=self.organisation.id,
+                        exc_info=exc_db,
+                    )
+                    continue
+
```

Comment:
For now I've opted to check if there is already a task available for the schedule. I want to optimize this a bit more by having 2 separate queries of schedules that have no cron expression and no tasks associated, and schedules that have a cron expression. Since for those two cases we need to create report task.

However since the PR https://github.com/minvws/nl-kat-coordination/pull/3840 is time-sensitive I've opted for this change atm. I'll make a ticket for this optimalisation and implement that so that it can be merged at a later  time.